src/core/ext/transport/chttp2/transport/chttp2_transport.c

```diff
@@ -2398,6 +2418,36 @@ void grpc_chttp2_config_default_keepalive_args(grpc_channel_args *args,
                 &args->args[i],
                 (grpc_integer_options){g_default_keepalive_permit_without_calls,
                                        0, 1});
+      } else if (0 ==
+                 strcmp(args->args[i].key, GRPC_ARG_HTTP2_MAX_PING_STRIKES)) {
+        g_default_max_ping_strikes = grpc_channel_arg_get_integer(
```

Comment:
Setting global state from transport construction is the wrong thing to do:
1. it's not thread safe (what if I create transports from two threads)
2. it's order dependent (max_ping_strikes now depends on the construction order of transports, which might change even due to network latency - meaning this could become a random number)

Better would be to hold these values as properties of the transport (preferably as some sub-struct... grpc_chttp2_ping_enforcement_args?)